Feature: Stars field

zhephyn opened the pull request:
# Description

This PR introduces a star system. For a rating, the total number of stars is 5 and a user can assign a rating of 0 to 5 on the edit and new pages. The selected ratings are then displayed on the show and index pages.  

Index Page:
<img width="1347" height="678" alt="image" src="https://github.com/user-attachments/assets/5598896e-0e48-481e-973e-dc4c0d4a67b1" />

Show Page:
<img width="1351" height="685" alt="image" src="https://github.com/user-attachments/assets/90fd04bd-d8aa-44a7-afcb-3fa3930b8a88" />

New page:
<img width="1162" height="677" alt="image" src="https://github.com/user-attachments/assets/23abdb18-620c-4363-afd3-e823ffdd945e" />

Edit page:
<img width="1298" height="673" alt="image" src="https://github.com/user-attachments/assets/e0497326-d3c1-4564-838c-93efce395da3" />


Fixes #1896 

# Checklist:
<!--
  Please go through the steps and complete them if they make sense (add tests if the change requires it, add to the docs, etc.)
  (Mark [x] inside the brackets)
-->

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

Paul-Bob reviewed (changes requested):
AMAZING @zhephyn, love it!

Let's just modify the SVG to use Avo's SVG helper with a Heroicon star SVG. Other than that, I think it's good to go. I'll do another in-depth review after the SVG change. Here is an example of [how we do it](https://github.com/avo-hq/avo/blob/main/app/components/avo/fields/password_field/edit_component.html.erb#L17).

zhephyn commented:
new changes for this are up!

Paul-Bob reviewed (changes requested):
Cool, now let's apply that changes to all the svgs

zhephyn commented:
Thanks for the heads up about next week. As for the PR, I've pushed the latest update .

Paul-Bob commented on app/components/avo/fields/common/stars_component.html.erb:
```diff
@@ -0,0 +1,6 @@
+<div class="rating-group" data-component="<%= component_name %>"> 
+  <% (1..@max).each do |i| %>
+    <% star_classes = "star #{'filled' if i <= (@value || 0)} inline-block mr-0.5" %>
+    <%= helpers.svg('heroicons/solid/star', class: star_classes) %>
```
Let's use the `class_names` helper for better readability, example: https://github.com/avo-hq/avo/blob/4-dev/app/components/avo/row_selector_component.html.erb#L8

Paul-Bob commented on app/components/avo/fields/common/stars_component.html.erb:
```diff
@@ -0,0 +1,6 @@
+<div class="rating-group" data-component="<%= component_name %>"> 
+  <% (1..@max).each do |i| %>
```
What is `i`? Let's rename it to something with meaning

Paul-Bob commented on app/components/avo/fields/stars_field/edit_component.html.erb:
```diff
@@ -0,0 +1,24 @@
+         data-max="<%= @field.max %>"
+         data-current-value="<%= @field.value || 0 %>">
+      <% (1..@field.max).each do |i| %>
+        <% star_classes = "star #{'filled' if i <= (@value || 0)} inline-block mr-0.5" %>
```
Let's use the `class_names` helper for better readability, example: https://github.com/avo-hq/avo/blob/4-dev/app/components/avo/row_selector_component.html.erb#L8

Paul-Bob commented on app/components/avo/fields/stars_field/edit_component.html.erb:
```diff
@@ -0,0 +1,24 @@
+      <% end %>
+    </div>
+  </div>
+<% end %>
```
Let's make sure all files have a new line at the end.

```suggestion
<% end %>

```

Paul-Bob reviewed (changes requested):
We're getting closer to merge this PR, I left some readability improvements suggestions

zhephyn commented:
Okay. Let me take a look and make the changes. 👍

zhephyn commented:
Everything on this one seems done on my end. I implemented suggestions from the last code review and added 2 tests to confirm that the feature works. 

Lmk in case of anything else.

Paul-Bob commented:
I'm at friendlyrb currently, I'll make a review on this PR as soon I get the chance. Thanks for looking into this!

zhephyn commented on app/components/avo/fields/common/stars_component.html.erb:
```diff
@@ -0,0 +1,7 @@
+<div class="rating-group" data-component="<%= component_name %>"> 
+  <% (1..@max).each do |rating_value| %>
+    <%= helpers.svg('heroicons/solid/star',
+                    class: class_names("star #{'filled' if rating_value <= (@value || 0)} inline-block mr-0.5")
```
Okay. Let me work on it and get back to you.
